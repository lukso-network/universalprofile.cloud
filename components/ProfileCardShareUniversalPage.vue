<script setup lang="ts">
type Props = {
  profile?: Profile
}

const props = defineProps<Props>()

const handleOpen = () => {
  navigateTo(universalPageProfileUrl(props.profile?.address), {
    external: true,
    open: {
      target: '_blank',
    },
  })
}
</script>

<template>
  <lukso-tooltip variant="white" text="universal.page" placement="left">
    <lukso-button
      size="small"
      variant="secondary"
      custom-class="px-0 overflow-hidden size-full"
      class="group size-[30px]"
      @click="handleOpen"
    >
      <img
        src="/images/universalpage-logo.png"
        alt=""
        class="h-4 grayscale transition group-hover:grayscale-0"
      />
    </lukso-button>
  </lukso-tooltip>
</template>
