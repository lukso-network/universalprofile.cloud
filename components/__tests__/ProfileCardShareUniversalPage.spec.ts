import { renderSuspended } from '@nuxt/test-utils/runtime'
import { screen } from '@testing-library/vue'
import { describe, expect, it, vi } from 'vitest'

import ProfileCardShareUniversalPage from '@/components/ProfileCardShareUniversalPage.vue'

describe('ProfileCardShareUniversalPage', () => {
  const profile: Profile = {
    address: '0x1234567890abcdef1234567890abcdef12345678',
  }

  it('should render', async () => {
    const component = await renderSuspended(ProfileCardShareUniversalPage, {
      props: { profile },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('should open in a new tab', async () => {
    const component = await renderSuspended(ProfileCardShareUniversalPage, {
      props: { profile },
    })

    const windowOpenSpy = vi.spyOn(window, 'open')

    ;(await component.findByRole('img')).click()

    expect(windowOpenSpy).toHaveBeenCalledWith(
      'https://universal.page/profiles/lukso/0x1234567890abcdef1234567890abcdef12345678',
      '_blank'
    )
  })
})
