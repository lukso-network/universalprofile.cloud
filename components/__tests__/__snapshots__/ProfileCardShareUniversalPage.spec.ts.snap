// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfileCardShareUniversalPage > should render 1`] = `
"<div id="test-wrapper">
  <lukso-tooltip variant="white" text="universal.page" placement="left">
    <lukso-button size="small" variant="secondary" custom-class="px-0 overflow-hidden size-full" class="group size-[30px]"><img src="/images/universalpage-logo.png" alt="" class="h-4 grayscale transition group-hover:grayscale-0"></lukso-button>
  </lukso-tooltip>
</div>"
`;
